influence_benchmark/retroactive_evaluator/openai_retroactive_evaluator.py

```diff
@@ -0,0 +1,130 @@
+    This class handles the evaluations for trajectories across a choice of metrics using the OpenAI API.
+    """
+
+    def __init__(
+        self,
+        run_path: Path,
+        backend_config: Dict,
+        metrics: List[str],
+        env_config_path: Optional[Path],
+        max_trajs_per_env: Optional[int],
+        backend: Optional[Backend] = None,
+    ):
+        """
+        Initialize the OpenAIRetroactiveEvaluator.
+
+        Args:
+            run_path (Path): Path to the run data.
+            backend_config (Dict): Configuration for the backend model.
+            metrics (List[str]): List of metrics to evaluate.
+            env_config_path (Optional[Path]): Path to environment configuration files for preference prompts.
+            max_trajs_per_env (int): Maximum number of randomly sampled trajectories per environment to evaluate.
+            backend (Optional[Backend]): An existing backend instance (optional).
+        """
+        self.backend_config = backend_config
+        self.backend = backend  # Optional pre-initialized backend
+        self.initialize_backend()
+        super().__init__(run_path, metrics, env_config_path, max_trajs_per_env)
+
+    def initialize_backend(self):
+        """
+        Initialize the OpenAI GPT backend.
+        """
+
+        assert LOADED_DOTENV, "API keys not loaded"
+        if self.backend is None:
+            assert (
+                "max_requests_per_minute" in self.backend_config
+            ), "max_requests_per_minute must be provided for GPT backend"
+            assert (
+                "max_tokens_per_minute" in self.backend_config
+            ), "max_tokens_per_minute must be provided for GPT backend"
+            self.backend = OpenAIBackend(**self.backend_config)
+
+    def _evaluate_transcripts(self, all_transcripts_with_env):
+        """
+        Evaluate transcripts using the OpenAI GPT backend.
+
+        Args:
+            all_transcripts_with_env (List[Tuple[int, Tuple[List[Dict[str, str]], str]]]):
+                A list of tuples containing the index and a tuple of (transcript, env_name).
+
+        Returns:
+            List[Tuple[int, Dict[str, float]]]: Evaluation results.
+        """
+        print("Sending requests to backend...")
+        vectorized_assessor = self.vectorized_assessor_for_backend_all_metrics(
+            self.backend, len(all_transcripts_with_env) * len(self.metrics)
+        )
+
+        indices = [item[0] for item in all_transcripts_with_env]
+        data = [item[1] for item in all_transcripts_with_env]
+
+        # Prepare all states with len(self.metrics) copies for each traj
+        states = [
+            self.prepare_state(transcript, env_name) for transcript, env_name in data for _ in range(len(self.metrics))
+        ]
+
+        results = []
+
+        responses = vectorized_assessor.get_response(states)
+        print(f"Obtained responses from GPT backend for all metrics.")
```

Comment:
Would be good to have a bit more info here. Maybe printing the number of requests that are being sent to the backend, and also tracking the time that the whole process took and printing that out. That's always nice/useful to have in the logs